src/core/sock/sockinfo_tcp.cpp

```diff
@@ -6017,15 +6022,16 @@ void tcp_timers_collection::add_new_timer(sockinfo_tcp *sock)
         return;
     }
 
-    // A reused time-wait socket wil try to add a timer although it is already registered.
-    if (m_sock_remove_map.find(sock) != m_sock_remove_map.end()) {
+    if (sock->is_timer_registered()) {
+        __log_warn("Trying to add timer twice for TCP socket %p", sock);
         return;
     }
 
     sock_list &bucket = m_p_intervals[m_n_next_insert_bucket];
     bucket.emplace_back(sock);
     m_sock_remove_map.emplace(sock, std::make_tuple(m_n_next_insert_bucket, --(bucket.end())));
     m_n_next_insert_bucket = (m_n_next_insert_bucket + 1) % m_n_intervals_size;
+    sock->set_timer_registered(true);
```

Comment:
Consider set it to true before putting in the bucket.
Not mandatory.

Reply:
As discussed, we leave it here as this should not affect functionality.